checkbox-support/checkbox_support/scripts/image_checker.py

```diff
@@ -3,20 +3,44 @@
 #
 # Written by:
 #   Patrick Chang <patrick.chang@canonical.com>
+#   Zhongning Li <zhongning.li@canonical.com>
 
 import argparse
 from os.path import exists
 from checkbox_support.snap_utils.system import on_ubuntucore
+from subprocess import DEVNULL, run
 
 
-def get_type():
+def get_type() -> str:
     """
     Return the type of image.
     """
     return "core" if on_ubuntucore() else "classic"
 
 
-def get_source():
+def has_desktop_environment(
+    desktop_packages=["ubuntu-desktop", "ubuntu-desktop-minimal"]
+) -> bool:
+    """
+    Returns whether there's a desktop environment
+    """
+    if on_ubuntucore():
+        # ubuntu core doesn't have a desktop env by default
+        return False
+
+    for package in desktop_packages:
+        if (
+            run(
+                ["dpkg", "-l", package], stdout=DEVNULL, stderr=DEVNULL
+            ).returncode
+            == 0
+        ):
+            return True
```

Comment:
Minor: I think this is easier to read (If you take this you need to import `contextlib.suppress`, `subprocess.check_call` and `subprocess.CalledProcessError`)
```suggestion
        with suppress(CalledProcessError):
            check_call(["dpkg", "-l", package])
            return True
```

Reply:
Ohh this looks like a cooler
```
try:
    ...
except:
    passs
```
Thanks for the suggestion! :D

Reply:
it is :)